File: backend/internal/entities/bus-trip.go
package entities

import (
	"time"

	"github.com/SamuelJacobsenB/project-coopcam-ifes/internal/types"
	"github.com/google/uuid"
)

type BusTrip struct {
	ID uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`

	Date      time.Time       `json:"date"`
	Period    types.Period    `gorm:"type:text" json:"period"`
	Direction types.Direction `gorm:"type:text" json:"direction"`
	Status    types.Status    `gorm:"type:text" json:"status"`
	
	Reports []BusTripReport `gorm:"foreignKey:BusTripID" json:"reports"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (BusTrip) TableName() string {
	return "bus_trips"
}

File: mobile/types/dtos/bus-trip.ts
import type { Direction, Period, Status } from "../";

export interface BusTripRequestDTO {
  date: Date;
  period: Period;
  direction: Direction;
}

export interface BusTripUpdateDTO {
  date: Date | null;
  period: Period | null;
  direction: Direction | null;
  status: Status | null;
}
